reflection.md

```diff
@@ -1,18 +1,71 @@
 # Reflection on implementing regular expressions of a DSL
 
 ## Which operators were easiest to implement and why?
+I found the binary operators and the star operator intuitive to implement, 
+since there were already case classes built for these operators such as Union and
+Concat. 
 
 ## Which operators were most difficult to implement and why?
+I spent a long time trying to think of how to implement the repetition operator,
+{n}. Just like other operators like <*> and <+>, I thought the curly braces 
+should be the function's name when defining it. However, I was having a problem because there's an integer between the curly braces and I didn't know how to 
+include that in the function's name, which made me wonder for a long time if I 
+can have a function's own parameter in the name of the function. It also took 
+me a long time to actually find out that there's a method 'apply' available for 
```

Comment:
Yeah, once you figure it out it shouldn't be too bad, but if you forget about this I can see why it's tricky.